src/rtt.cpp

```diff
@@ -0,0 +1,590 @@
+        Nan::ThrowError(message);
+        return;
+    }
+
+    resetLog();
+
+    auto argumentCount = 0;
+    RTTBaton *baton = nullptr;
+
+    try
+    {
+        baton = parse(info, argumentCount);
+
+        v8::Local<v8::Function> callback = Convert::getCallbackFunction(info[argumentCount]);
+        baton->callback = new Nan::Callback(callback);
+        argumentCount++;
+
+        if (info.Length() > argumentCount)
+        {
+            argumentCount = CUSTOM_ARGUMENT_PARSE_ERROR;
+            std::ostringstream errorStringStream;
+            errorStringStream << "Too many parameters. The function " << baton->name << " do not take " << info.Length() << " parameters.";
+            throw errorStringStream.str();
+        }
+    }
+    catch (std::string error)
+    {
+        if (baton != nullptr)
+        {
+            delete baton;
+        }
+
+        auto message = ErrorMessage::getTypeErrorMessage(argumentCount, error);
+        Nan::ThrowTypeError(message);
+
+        return;
+    }
+
+    // This is a check that there exists a returnfunction when there are more returns
+    // than just err. If this shows up in production, it is due to missing return function
+    if (ret == nullptr &&
+        baton->returnParameterCount > 0)
+    {
+        auto message = ErrorMessage::getErrorMessage(1, rtt_err_map, std::string("The return function has more than one parameter and is required for this function, but is missing"));
+        Nan::ThrowError(message);
+        return;
+    }
+
+    log("===============================================\n");
+    log("Start of ");
+    log(baton->name);
+    log("\n");
+    log("===============================================\n");
+
+
+    baton->executeFunction = execute;
+    baton->returnFunction = ret;
+
+    uv_queue_work(uv_default_loop(), baton->req, ExecuteFunction, reinterpret_cast<uv_after_work_cb>(ReturnFunction));
+}
+
+void RTT::ExecuteFunction(uv_work_t *req)
+{
+    auto baton = static_cast<RTTBaton *>(req->data);
+
+    RTTErrorcodes_t executeError = baton->executeFunction(baton);
+
+    if (executeError != RTTSuccess)
+    {
+        baton->result = executeError;
+    }
+}
+
+void RTT::ReturnFunction(uv_work_t *req)
+{
+    Nan::HandleScope scope;
+
+    auto baton = static_cast<RTTBaton *>(req->data);
+    std::vector<v8::Local<v8::Value> > argv;
+
+    argv.push_back(ErrorMessage::getErrorMessage(baton->result, rtt_err_map, baton->name, logMessage, baton->lowlevelError));
+
+    if (baton->result != errorcode_t::JsSuccess)
+    {
+        for (uint32_t i = 0; i < baton->returnParameterCount; i++)
+        {
+            argv.push_back(Nan::Undefined());
+        }
+    }
+    else
+    {
+        if (baton->returnFunction != nullptr)
+        {
+            std::vector<v8::Local<v8::Value> > vector = baton->returnFunction(baton);
```

Comment:
Using variable names matching standard typenames I don't suggest in general, misleading to read.

Reply:
Will fix.